igv_remote/_core.py

```diff
@@ -54,25 +48,27 @@ def _set_saveopts(self, img_dir, img_basename, img_init_id=0) :
         self._img_basename = img_basename
         self._img_id = img_init_id
     
-    def _set_viewopts(self, view_type, viewaspairs, sort):
+    def _set_viewopts(self, view_type,  sort):
         if view_type not in ['squished', 'collapsed', 'expanded']:
             raise ValueError("view_type must be one of [squished, collapsed, expanded]")
         self._view_type = view_type
-        self._viewaspairs = viewaspairs
         self._sort = sort
 
     def _connect(self, host="127.0.0.1", port=60151):
+        assert self.sock is None
+        self.sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
         self.sock.connect((host, port))
         print("socket initialized")
     
     def _new(self):
         self._send("new ")
 
     def _send(self, cmd):
-        s = self.sock
+        assert self.sock is not None
+        s=self.sock
         cmd = cmd + '\n'
         s.send(cmd.encode('utf-8'))
-        return s.recv(2000).decode('utf-8').rstrip('\n')
```

Comment:
Maybe make a return value optional here? For debugging purposes it can be good to have the script echo back IGV's response.

Reply:
Do you mean something like:
```
if verbose:
    return s.recv(2000).decode('utf-8').rstrip('\n')
```

Reply:
Exactly. Or just print it; not sure if the calling function needs to do anything with the return value.